api/src/transactions.rs

```diff
@@ -66,6 +66,7 @@ impl SelfActionBuilder {
 #[derive(Debug, Clone)]
 pub struct ConstructTransaction {
     pub tr: PrepopulateTransaction,
+    pub deferred_error: Option<ArgumentSerializationError>,
 }
```

Comment:
1. Should it be an enum or a field of `Result<PrepopulatedTransaction, ArgumentSerializationError>` type?
2. May I ask you to use a more descriptive name than `tr`? 🙏 
3. Should we make the fields non-pub?
4. With `IntoAccountId`, the account ids will be also failable, and it would be nice to defer those errors as well, so consider making the error type more generic